### DISCUSSION TITLE
`workspace` added for syncVersions

### DISCUSSION THREAD

**Discussion by jwo** [jscutlery/semver#577] [General]
Hi there!

Assumption: It looks like when I want to run in syncVersions mode, I need to run the  install with `--syncVersions`

This then creates  this in my workspace.json, and I can then run `npx nx workspace:version`

```json
 "workspace": {
      "root": ".",
      "targets": {
        "version": {
          "executor": "@jscutlery/semver:version",
          "options": {
            "syncVersions": true,
            "preset": "conventional"
          }
        }
      }
    }
```

Complication: I then have to update my build script to exclude the workspace:
```
nx run-many --target=build --exclude=workspace --all
```

My question: Is there a way around having to exclude workspace with my run-many? Can we configure symver to sync versions without having a workspace project added to my workspace.json ?

**Comment by anthodb** [comment]
Wow.. this should be documented, I spent almost 2 hours trying to figure out how to setup the syncVersion... Thanks for this Post! 
But I'm not sure about your issue, why do you have conflicts with the build script? The "workspace" doesn't include this target? So it should not run in "run-many" command..

[EDIT]: indeed, all run-many commands seem to break after this setup is used..

**Reply by jwo** [reply]
Yeah, it might be a problem with the run-many command really. I'm not sure -- I agree with your thinking though: since workspace didn't include the target, it shouldn't care.

**Reply by anthodb** [reply]
I think this is related to the way the target is set in the workspace.json, actually in the new version of workspace.json, we are using path to projects (version2) and here it's using the project node inside the file (version1) which is not recommended anymore by NX team : https://nx.dev/configuration/projectjson. There should be a way to get rid of the "workspace" project and use some global executor to run the "version script".. but I didn't find a way yet..
